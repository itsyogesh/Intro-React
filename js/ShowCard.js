import React, { Component } from 'react'

class ShowCard extends Component {
  render() {
    const { poster, title, year, description } = this.props.show
    return (
      <div className='show-card'>
        <img src={`/public/img/posters/${poster}`} />
        <div>
          <h3>{title}</h3>
          <h4>({year})</h4>
          <p>{description}</p>
        </div>
      </div>
    )
  }
}

export default ShowCard
