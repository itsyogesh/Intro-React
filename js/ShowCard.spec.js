import React from 'react'
import ShowCard from './ShowCard'
import renderer from 'react-test-renderer'
import preload from '../public/data.json'

test('show-card snapshot test', () => {
  const component = renderer.create(<ShowCard show={preload.shows[0]} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
