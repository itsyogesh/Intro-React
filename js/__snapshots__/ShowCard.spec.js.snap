exports[`test show-card snapshot test 1`] = `
<div
  className="show-card">
  <img
    src="/public/img/posters/hoc.jpg" />
  <div>
    <h3>
      House of Cards
    </h3>
    <h4>
      (
      2013â€“
      )
    </h4>
    <p>
      A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him.
    </p>
  </div>
</div>
`;
